### ISSUE TITLE
[VersionControl] Error committing changes of removed files (Git)

### ISSUE THREAD

**Issue by jsuarezruiz** [mono/monodevelop#7119]
After deleting files, and try to commit the changes.
```
ERROR [2019-02-11 08:45:15Z]: Version control operation failed: 
LibGit2Sharp.EmptyCommitException: No changes; nothing to commit.
  at LibGit2Sharp.Repository.Commit (System.String message, LibGit2Sharp.Signature author, LibGit2Sharp.Signature committer, LibGit2Sharp.CommitOptions options) [0x000dd] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/external/libgit2sharp/LibGit2Sharp/Repository.cs:1084 
  at LibGit2Sharp.RepositoryExtensions.Commit (LibGit2Sharp.IRepository repository, System.String message, LibGit2Sharp.Signature author, LibGit2Sharp.Signature committer) [0x00001] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/external/libgit2sharp/LibGit2Sharp/RepositoryExtensions.cs:293 
  at MonoDevelop.VersionControl.Git.GitRepository.OnCommit (MonoDevelop.VersionControl.ChangeSet changeSet, MonoDevelop.Core.ProgressMonitor monitor) [0x000dc] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/GitRepository.cs:867 
  at MonoDevelop.VersionControl.Repository.Commit (MonoDevelop.VersionControl.ChangeSet changeSet, MonoDevelop.Core.ProgressMonitor monitor) [0x0005d] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/Repository.cs:593 
  at MonoDevelop.VersionControl.CommitCommand+CommitWorker.Run () [0x00081] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/CommitCommand.cs:70 
  at MonoDevelop.VersionControl.VersionControlTask.<BackgroundWorker>b__13_0 () [0x00000] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/VersionControlTask.cs:51 
  at System.Threading.Tasks.Task.InnerInvoke () [0x0000f] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-10/external/bockbuild/builds/mono-x64/external/corert/src/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs:2476 
  at System.Threading.Tasks.Task.Execute () [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-10/external/bockbuild/builds/mono-x64/external/corert/src/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs:2319
```

> VS bug [#789312](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/789312)

**Comment by jsuarezruiz** [comment]
Is the same as: https://developercommunity.visualstudio.com/content/problem/322119/git-version-control-operation-filed.html

**Comment by jsuarezruiz** [comment]
Is the same as: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/663199

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-25] Comment by **Lluis Sanchez Gual**




Fixed in version 8.0.0.2893 (release-8.0)


Pull Request #365 merged by: Vsevolod Kukol  
Author: xamarin  
Commit: 6395a1258bf6fd1f84eeb1be386ca435eba50e57 (xamarin/monodevelop)
